src/menu-button/_MenuButton.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import Menu from 'terra-menu';
+import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
+import Button from 'terra-button';
+
+import styles from './MenuButton.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Button text
+   */
+  text: PropTypes.string.isRequired,
+
+  /**
+   * menu items
+   */
+  items: PropTypes.arrayOf(PropTypes.string).isRequired,
+
+  /**
+   * Selected menu item
+   */
+  selectedKey: PropTypes.string,
+
+  /**
+   * On change callback
+   */
+  onChange: PropTypes.func.isRequired,
+};
+
+const MenuButton = ({
+  text, items, selectedKey, onChange,
+}) => {
+  const buttonRef = useRef(null);
+  const [state, setState] = useState(false);
```

Comment:
Could rename to [isOpen, setIsOpen] to be more clear about what this state represents.